src/Mammon/Workflows/Activities/AssignCostCentreActivity.cs

```diff
@@ -0,0 +1,21 @@
+﻿namespace Mammon.Workflows.Activities;
+
+public class AssignCostCentreActivity : WorkflowActivity<AssignCostCentreActivityRequest, bool>
+{
+    public override async Task<bool> RunAsync(WorkflowActivityContext context, AssignCostCentreActivityRequest input)
+    {
+        ArgumentNullException.ThrowIfNull(input);
+
+        //get cost centre costs
+        var costCentreCosts = await ActorProxy.DefaultProxyFactory.CallActorWithNoTimeout<IResourceActor, IDictionary<string, double>>(input.ResourceActorId, "ResourceActor", async (p) => await p.AssignCostCentreCosts()) 
+            ?? throw new InvalidOperationException($"Unexpected result returned for ${input.ResourceId} - AssignCostCentreCosts");
+
+        foreach (var (costCentre, costCentreCost) in costCentreCosts)
+        {            
+            //send them to cost centre actors
+            await ActorProxy.DefaultProxyFactory.CallActorWithNoTimeout<ICostCentreActor>($"{input.ReportId}_{costCentre}", "CostCentreActor", async (p) => await p.AddCost(input.ResourceId, costCentreCost));
```

Comment:
should we rename this `AddCost` to `AddCostAsync`?

Reply:
and `AssignCostCentreCosts`? maybe if you only have async methods, you don't need to differentiate :| not sure what's the convention

Reply:
let me do it... I kinda belive async suffix is artificial (visible from signature)  but lot of people need that to sleep better :D